reefperf/cloud_driver.py

```diff
@@ -11,40 +14,41 @@ class CloudDriver(object):
     def create_node(self, cloud_node_cfg):
         pass
 
+
 # LC is a short for apache libcloud. It means that
 # all classes with prefix LC using this libcloud for operating on cloud nodes
 
 
 class LCCloudScaleDriver(CloudDriver):
-    SSH_KEY_LENGHT = 4096
+    SSH_KEY_LENGTH = 4096
 
     def __init__(self, credentials):
         driver_cls = libcloud.get_driver(
             libcloud.DriverType.COMPUTE,
-            libcloud.DriverType.COMPUTE.CLOUDSCALE
+            libcloud.DriverType.COMPUTE.CLOUDSCALE,
         )
         self._driver = driver_cls(credentials['api_token'])
-        self._images = {
-            image.id: image for image in self._driver.list_images()
-        }
-        self._sizes = {
-            size.id: size for size in self._driver.list_sizes()
-        }
+
+    @property
+    @lru_cache()
+    def images(self):
+        return {image.id: image for image in self._driver.list_images()}
+
+    @property
+    @lru_cache()
+    def sizes(self):
+        return {size.id: size for size in self._driver.list_sizes()}
 
     def create_node(self, cloud_node_cfg):
```

Comment:
I think this method is very special - instead of accepting mandatory arguments like any other method, it accepts a container and references some keys from it. Does it have to be special? Shouldn't it be the responsibility of the caller to pass the arguments to the method?

Reply:
This method is very important for whole reeperf, but it is invoked only by internals of Theater class, which prepare argument for `create_node` .  Final container which is passed to method is a mix of data from two config files: app_deployment_config and node_types_config. So, I don't see, any other solution. Do you have idea, for better solution?